### PR TITLE
Build and upload APKs as artifacts

### PR DESCRIPTION
**Pull request by PatrykMis**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -75,7 +80,9 @@ android {
         release {
             isMinifyEnabled = false
 
-            signingConfig = signingConfigs.getByName("release")
+            if (useKeystoreProperties) {
```
Comment (Irineu333) [inline app/build.gradle.kts]: Is this change really necessary? For now, we just want to build as debug.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Build application
+
+on: [pull_request, push]
```
Comment (Irineu333) [inline .github/workflows/build.yml]: This way, it will run the action twice. This also used to happen with Validate Gradle Wrapper, but I fixed it.

**Comment by Irineu333** [comment]
I looked at the artifacts and the apk is not listed there: "No files were found with the provided path: build/outputs/apk/debug/app-debug.apk build/outputs/apk/release/app-release-unsigned.apk. No artifacts will be uploaded."

I believe the path is wrong, the apk will be generated in the application module. That is, in 'app/build/outputs/apk/debug/app-debug.apk'.

**Review by Irineu333 — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts (hunk 1 above)
Comment (PatrykMis) [reply app/build.gradle.kts]: This is nessesary, because:

1. Otherwise the build will fail
2. We can obtain signed debug and unsigned release (in artifacts) which can be signed by whoever wants with apktool

Alternatively, we can skip this and then build with command `./gradlew build -x assembleRelease --no-daemon` on ci, though I prefer to have both debug and release, even if it's unsigned.

Comment (Irineu333) [reply app/build.gradle.kts]: Understood. But why do we want to generate a release APK in a pull request to develop? I think the correct time to generate it would be at the creation of a release, when merging into master.

**Comment by Irineu333** [comment]
> I looked at the artifacts and the apk is not listed there: "No files were found with the provided path: build/outputs/apk/debug/app-debug.apk build/outputs/apk/release/app-release-unsigned.apk. No artifacts will be uploaded."
> 
> I believe the path is wrong, the apk will be generated in the application module. That is, in 'app/build/outputs/apk/debug/app-debug.apk'.

Resolved

File: .github/workflows/build.yml (hunk 2 above)
Comment (Irineu333) [reply .github/workflows/build.yml]: I think is enough run in pull requests. Or, if you want to run in push, run only for pushes for develop (although it's blocked for push, just like master). Otherwise, it will flood with logs, just like what was happening with the Validate Gradle Wrapper.

**Review by Irineu333 — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts (hunk 1 above)
Comment (PatrykMis) [reply app/build.gradle.kts]: If you really don't want to upload release apk as an artifact, we can change it in the workflow. But this `is` instruction is needed anyway because the project is not building properly if keystore.properties is absent. Let users build whatever they want either signed or unsigned.

Comment (Irineu333) [reply app/build.gradle.kts]: Isn't it enough to run a `buildDebug`? Or `assembleDebug`, since we don't have unit tests yet. I fixed the issue where the project wouldn't build without the `keystore.properties` even in debug mode.

Comment (PatrykMis) [reply app/build.gradle.kts]: Ok, we can run just `assembleDebug`, although it will skip lint.

Alternatively, we can `./gradlew check assembleDebug --no-daemon`.

Comment (Irineu333) [reply app/build.gradle.kts]: I haven't set up any lint in the project yet

Comment (PatrykMis) [reply app/build.gradle.kts]: You don't have to, AGP does it in basic way. For example, if you add some strings in `values/strings.xml` and don't translate it, it throws an error.

Lint reports are at `app/build/reports/lint-results-debug.html`; at the current stage of development it doesn't show any error. Now two wornings from develop branch shows obsolete dependency which are updated in #116.

<https://proandroiddev.com/effectively-using-android-lint-in-existing-projects-eacff100a919>

To sum up, lint (`./gradlew check` or `./gradlew lint` is capable of identifying most of trivial errors such as untranslated strings, some deprecated APIs, images without alternative text etc. Further more, if lint throws an error and `lint-baseline.xml` is not present, or other annotations in the code, the `./gradlew build` will fail and won't create apk. That's `lint.abortOnError = false` exists in some projects build.gradle.kts files.

Comment (Irineu333) [reply app/build.gradle.kts]: Man, that's awesome. I'm learning a lot from you.

Comment (Irineu333) [reply app/build.gradle.kts]: Well, we can manually run the steps that assembleDebug skips, right?

Comment (Irineu333) [reply app/build.gradle.kts]: Well, I leave the decision up to you. But if you choose to keep it, I suggest calling `keystorePropertiesFile.canRead()` directly and modifying `Project.properties` extension to receive `keystorePropertiesFile` instead of the name

Comment (PatrykMis) [reply app/build.gradle.kts]: I'm learning a lot from you, too. Did as you suggested in recent commits.

**Review by Irineu333 — APPROVED** [review]
